x/group/keeper/msg_server.go

```diff
@@ -663,6 +663,10 @@ func (k Keeper) WithdrawProposal(goCtx context.Context, msg *group.MsgWithdrawPr
 		return nil, err
 	}
 
+	if err := ctx.EventManager().EmitTypedEvent(&group.EventProposalFinalized{ProposalId: proposal.Id, Status: proposal.Status}); err != nil {
```

Comment:
This is probably not necessary, as this event will be emitted for withdrawn proposal in `TallyProposalsAtVPEnd`.
If it is, let's use `EmitTypedEvents` instead and combine this line and the one below.